src/webnn_native/ops/Squeeze.h

```diff
@@ -35,6 +37,57 @@ namespace webnn_native { namespace op {
             return graph->AddSqueeze(this);
         }
 
+        MaybeError CalculateShape() override {
+            auto inputShape =
+                mInputs[0]->Shape().empty() ? std::vector<int32_t>{1} : mInputs[0]->Shape();
+            auto inputRank = inputShape.size();
+
+            std::unordered_set<int32_t> axesSqueezed;
+            if (mAxes.empty()) {
+                for (size_t i = 0; i < inputRank; ++i) {
+                    if (inputShape[i] == 1) {
+                        axesSqueezed.insert(i);
+                    }
+                }
+            } else {
+                for (const auto& axis : mAxes) {
+                    axesSqueezed.insert(axis);
+                }
+            }
+
+            std::vector<int32_t> outputShape;
+            for (size_t i = 0; i < inputRank; ++i) {
+                if (axesSqueezed.find(i) == axesSqueezed.end()) {
+                    outputShape.push_back(inputShape[i]);
+                }

```

Comment:
When squeeze the axis i, should it check whether `inputShape[i] == 1`?

Is it better to write the algorithm like?

1. Iterate all axes, for each axis:
    1. If the size is 1, then:
         1.  If `axesToSqueeze` contains this axis or `axesToSqueeze` is empty, then drop this axis and go to the next iteration.
     1. Keep this axis.

Reply:
My codes here is to handle axes with some duplicated components or with a unexpected order.
`When squeeze the axis i, should it check whether inputShape[i] == 1?` is right, but it is not needed by calculating the output shape I think. Now I move all validations before constructing a new operator base. We can discuss for them in details.  @fujunwei @huningxin

Reply:
Thanks, I have checked `When squeeze the axis i, should it check whether inputShape[i] == 1?`

Reply:
now the code looks good.